### PR TITLE
FABN-1506: JSDoc for block, contract and commit events

### PR DESCRIPTION
**Pull request by bestbeforetoday**

### CODE REVIEW

File: fabric-network/src/events.ts [hunk 1]
```diff
@@ -8,34 +8,34 @@ import { FilteredBlock, FilteredTransaction, Endorser, EventInfo } from 'fabric-
 import Long = require('long');
 
 export interface BlockEvent {
-	getType(): 'filtered' | 'full' | 'private';
-	getBlockNumber(): Long;
+	readonly type: 'filtered' | 'full' | 'private';
```
Comment (heatherlp) [inline fabric-network/src/events.ts]: is it worth updating the index.js jsdoc file to say that these properties are readonly? or is that pointless?

**Review by heatherlp — COMMENTED** [review]
We've got some lines ending in full stops and other not, so we're not consistent everywhere, but that's not a big deal right now

File: fabric-network/index.js [hunk 2]
```diff
@@ -226,56 +226,145 @@
...
+
+
+/**
+ * Event representing a transaction processed within a block.
+ * @interface TransactionEvent
+ * @memberof module:fabric-network
+ * @property {string} transactionId The ID of the transaction this event represents.
+ * @property {string} status The status of this transaction.
+ * @property {boolean} isValid Whether this transaction was successfully committed to the ledger. <code>true</code> if
+ * the transaction was commited; otherwise <code>false</code>. The status will provide more a more specific reason why
```
[the remark this answers is not in the document]
Comment (bestbeforetoday) [reply fabric-network/index.js]: Should be fixed

File: fabric-network/index.js [hunk 3]
```diff
@@ -226,56 +226,145 @@
...
+
+/**
+ * Get the parent block event for this event.
+ * @method TransactionEvent#getBlockEvent
+ * @memberof module:fabric-network
+ * @returns {module:fabric-network.BlockEvent} A block event.
+ */
+
+/**
+ * Get the contract events emitted this transaction.
```
[the remark this answers is not in the document]
Comment (bestbeforetoday) [reply fabric-network/index.js]: Should be fixed

File: fabric-network/index.js [hunk 4]
```diff
@@ -312,10 +401,10 @@
  * Add a listener to receive transaction commit and peer disconnect events for a set of peers.
  * @method Network#addCommitListener
  * @memberof module:fabric-network
- * @param {module:fabric-network.Network~CommitListener} listener A transaction commit listener callback function.
+ * @param {module:fabric-network.CommitListener} listener A transaction commit listener callback function.
```
[the remark this answers is not in the document]
Comment (bestbeforetoday) [reply fabric-network/index.js]: The initial `listener` is the parameter name, not part of the description

**Review by heatherlp — APPROVED** [review]

**Pull request merged**